src/Stratis.Bitcoin.Features.Consensus.Tests/TimeSyncBehaviorAdjustedTimeTest.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.Consensus.Tests
+{
+    /// <summary>
+    /// Tests of <see cref="TimeSyncBehavior"/> and <see cref="TimeSyncBehaviorState"/> classes.
+    /// </summary>
+    public partial class TimeSyncBehaviorAdjustedTimeTest
+    {
+        /// <summary>
+        /// Time data is added this this state. The combination of this and datetimeprovider are what is under test.
+        /// </summary>
+        private TimeSyncBehaviorState timesyncBehaviourState;
+
+        /// <summary>
+        /// This provides the adjusted time ready for assertion. The combination of this and timesyncBehaviorState are what is under test.
+        /// </summary>
+        private IDateTimeProvider dateTimeProvider;
+        
+        /// <summary>
+        /// This is the result of the operation under test. It is only rounded so that assertions are easier, a future improvement would be to not use UtcNow directly in the code, allowing it to be replaced at test time.
+        /// </summary>
+        private int roundedOffset;
+
+        /// <summary>
+        /// Checks that <see cref="TimeSyncBehaviorState.AddTimeData(IPAddress, TimeSpan, bool)"/>
+        /// uses outbound samples as a priority over inbound as they are less likely to be malicious.
+        /// This means that even when there are many inbound connections, only the first ones are considered until there are enough outbound nodes. 
+        /// Eg only 9 inbound per 1 outbound.
+        /// </summary>
+        [Fact]
+        public void AddTimeData_WithLargeSampleSetOfInboundTimeManipulatorsAndLowSampleSetOfOutbound_GetsOverridenByOutboundSamples()
+        {
+            given_an_empty_time_sync_behaviour_state();
```

Comment:
Everywhere here and in the next test use `this.`

Reply:
for methods in any other context I agree, however with the intention of this being readable tests, this would be an exception to that rule that polutes it and makes it less readable.

Reply:
I agree that it affects the readability in some cases like that one and I personally don't use it in my projects but that's the policy- always use `.this`

Reply:
I have added this. and capitalised the Given, When and Then's now to conform.